docs/ml_system_design_doc.md

```diff
@@ -0,0 +1,201 @@
+
+#### 1.3. Что входит в скоуп проекта/итерации, что не входит   
+
+- На закрытие каких БТ подписываемся в данной итерации: Создание базовой мультиагентной архитектуры (конкретный набор агентов будет определен в процессе проектирования), генерация текстовых описаний мира и персонажей, формирование простых сюжетных линий с возможностью базового взаимодействия пользователя.
+- Что не будет закрыто: Генерация изображений локаций и персонажей, сложные ветвящиеся сюжеты, полноценная система диалогов.
+- Описание результата с точки зрения качества кода и воспроизводимости решения: Модульная архитектура с четким разделением ответственности между агентами, документированный код, возможность легкого расширения функциональности в будущих итерациях.
+- Описание планируемого технического долга: Оптимизация производительности для обработки более сложных сценариев, интеграция с системами визуализации, доработка механизмов для более естественного восприятия вводимых пользователем команд и запросов.
+
+#### 1.4. Предпосылки решения  
+
+- Описание всех общих предпосылок решения: Мы будем использовать предобученные языковые модели для генерации текстового контента, с возможностью дообучения на специализированных текстах. Архитектура системы будет опираться на концепцию мультиагентного взаимодействия, где AMI выступает в роли Главного агента, который напрямую взаимодействует с пользователем и координирует работу других специализированных агентов. Для хранения и управления состоянием истории будем использовать графовую базу данных для отслеживания взаимосвязей между элементами мира.
+
+### 2. Методология  
+
+#### 2.1. Постановка задачи  
+
+- С технической точки зрения нам необходимо создать систему генерации текста специфического домена, обеспечить когерентность и целостность создаваемого контента через мультиагентную архитектуру, реализовать механизмы для адаптивной генерации сюжета на основе действий пользователя. Основные компоненты включают системы генерации мира, персонажей и сюжета, а также механизмы обработки естественного языка для взаимодействия с пользователем.
+
+#### 2.2. Блок-схема решения  
+
+- Базовая архитектура AMI включает следующие компоненты:
+  1. Интерфейс пользователя (UI) - для ввода команд и отображения генерируемого контента
+  2. AMI (Главный агент) - напрямую взаимодействует с пользователем и координирует работу специализированных агентов
+  3. Мультиагентная система - набор специализированных агентов, состав которых будет определен в процессе исследований и проектирования
+  4. База данных - хранит состояние мира, персонажей и сюжета
+
+- Бейзлайн: использование единой модели для генерации всего контента без разделения на агентов
+- MVP: мультиагентная система с разделением ответственности между специализированными агентами
+
+#### 2.3. Этапы решения задачи  
+
+*Этап 1 - Подготовка данных*
+
+- Данные и сущности, на которых будет обучаться модель машинного обучения:
```

Comment:
Do you really want to train (fine-tune) model on this data or just use it as in-context learning examples? 
How much data do you plan to collect?